### PR TITLE
Issue #94

### LINKED ISSUE #94: Expose database replace in UI
- [ ] rename files to more explicit eg 'file_search' 'file_replace' 'db_search' db_replace
- [ ] run a search, make sure it has the placeholder column (existing code)
- [ ] add some replace items to the csv
- [ ] upload the same file again
- [ ] unlock the session
- [ ] if no replace then skip the record
- [ ] if there is then do the replace on the record and store if the content ended up changing or not
- [ ] have a progress bar and do the process inline in the web process not in a task (we may do this async in future)
- [ ] when finished produce a summary of skipped vs replaced vs unmatched

### PR DESCRIPTION
**Pull request by petersistrom**
Resolves #94

### CODE REVIEW

File: db_replace.php [hunk 1]
```diff
@@ -43,12 +48,23 @@
 echo $OUTPUT->header();
 if ($form->is_cancelled()) {
     redirect($redirect);
-} else if ($csvcontent = $form->get_file_content('csvfile')) {
-    helper::handle_replace_csv($csvcontent);
-} else {
+} else if (!isset($CFG->allowuireplace) || !$CFG->allowuireplace) {
```
Comment (brendanheywood) [inline db_replace.php]: CFG->allowuireplace needs to be namespaced eg in config.php:

$CFG->forced_plugin_settings['tool_advancedreplace']['allowuireplace'] = 'duck';

and in code its normal get_config()

File: db_replace.php [hunk 2]
```diff
@@ -43,12 +48,23 @@
...
+    echo $OUTPUT->confirm(get_string('replacecheck', 'tool_advancedreplace'), $deletebutton, $returnurl);
+} else if ($confirm && isset($csvpostcontent)) {
+    helper::handle_replace_csv($csvpostcontent);
+}else {
     // Display form.
     echo $OUTPUT->heading(get_string('replacepageheader', 'tool_advancedreplace'));
     $form->display();
 }
 
+
```
Comment (brendanheywood) [inline db_replace.php]: extra whitespace

File: lang/en/tool_advancedreplace.php [hunk 3]
```diff
@@ -92,6 +92,8 @@
 $string['replacepageheader'] = 'Replace text stored in the DB';
 $string['replacepagename'] = 'Replace strings in the Database';
 $string['replace'] = 'Replace';
+$string['replacecheck'] = 'Are you sure you want to replace strings in the Database using the uploaded csv file?';
```
Comment (brendanheywood) [inline lang/en/tool_advancedreplace.php:95]: ideally this shows vaguely what it is about to do, eg 'we are about to replace 12,456 records - are you sure?'

File: db_replace.php (hunk 2 above)
Comment (petersistrom) [reply db_replace.php]: removed

File: db_replace.php (hunk 1 above)
Comment (petersistrom) [reply db_replace.php]: Update with name and to use to get_config()

**Review by brendanheywood — APPROVED** [review]

**Pull request merged**